packages/components/pagination/src/pagination.ts

```diff
@@ -152,20 +152,37 @@ export default defineComponent({
     })
 
     const innerPageSize = ref(
-      isAbsent(props.defaultPageSize) ? 10 : props.defaultPageSize
+      (() => {
+        let pageSize = 10
+        if (!isAbsent(props.defaultPageSize)) {
+          pageSize = props.defaultPageSize
+        }
+        if (!isAbsent(props.pageSize)) {
+          pageSize = props.pageSize
+        }
+        return pageSize
+      })()
+    )
+
+    watch(
+      () => props.pageSize,
+      () => {
```

Comment:
I think we can get the value here by:

```ts
watch(() => props.pageSize, (pageSize) => {
  // do all the stuff
})
```

Reply:
**Question:** if I passed a`pageSize` with a valid variable, then I pass another one which makes `isAbsent` true, the `innerPageSize.value` will not be updated, which means, page size is still not reactive, is this intended?

Reply:
Yeah,I add this logic as the result of the comment block on the isAbsent variable. It told that using the legal input is user's responsibility.

Reply:
Should we throw a error or a warning message during isAbsent equals to true? 
I am curious about why using the isAbsent check logic instead of using the vue native props type check?

Reply:
I think it wants to make the component work no matter what the user's input is, so it used `isAbsent` to determine whether the attribute has been given, I think maybe you were right, it should always be the users responsibility, not ours. Let's use your way here.

Loop the original author here @ioslh